src/measure_theory/group/add_circle.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import measure_theory.integral.periodic
+import data.zmod.quotient
+
+/-!
+# Measure-theoretic results about the additive circle
+
+The file is a place to collect measure-theoretic results about the additive circle.
+
+## Main definitions:
+
+ * `add_circle.closed_ball_ae_eq_ball`: open and closed balls in the additive circle are almost
+   equal
+ * `add_circle.is_add_fundamental_domain_of_ae_ball`: a ball is a fundamental domain for rational
+   angle rotation in the additive circle
+
+-/
+
+open set function filter measure_theory measure_theory.measure metric
+open_locale measure_theory pointwise big_operators topological_space ennreal
+
+namespace add_circle
+
+variables {T : ℝ} [hT : fact (0 < T)]
+include hT
+
+lemma closed_ball_ae_eq_ball {x : add_circle T} {ε : ℝ} :
+  closed_ball x ε =ᵐ[volume] ball x ε :=
+begin
+  cases le_or_lt ε 0 with hε hε,
+  { rw [ball_eq_empty.mpr hε, ae_eq_empty, volume_closed_ball,
+      min_eq_right (by linarith [hT.out] : 2 * ε ≤ T), ennreal.of_real_eq_zero],
+    exact mul_nonpos_of_nonneg_of_nonpos zero_le_two hε, },
+  { suffices : volume (closed_ball x ε) ≤ volume (ball x ε),
+    { exact (ae_eq_of_subset_of_measure_ge ball_subset_closed_ball this measurable_set_ball
+        (measure_ne_top _ _)).symm, },
+    have : tendsto (λ δ, volume (closed_ball x δ)) (𝓝[<] ε) (𝓝 $ volume (closed_ball x ε)),
+    { simp_rw volume_closed_ball,
+      refine ennreal.tendsto_of_real (tendsto.min tendsto_const_nhds $ tendsto.const_mul _ _),
+      convert (@monotone_id ℝ _).tendsto_nhds_within_Iio ε,
+      simp, },
+    refine le_of_tendsto this (mem_nhds_within_Iio_iff_exists_Ioo_subset.mpr ⟨0, hε, λ r hr, _⟩),
+    exact measure_mono (closed_ball_subset_ball hr.2), },
+end
+
+lemma is_add_fundamental_domain_of_ae_ball (I : set $ add_circle T)
+  (u x : add_circle T) (hu : is_of_fin_add_order u)
+  (hI : I =ᵐ[volume] ball x (T / (2 * add_order_of u))) :
+  is_add_fundamental_domain (add_subgroup.zmultiples u) I :=
+is_add_fundamental_domain.mk_of_measure_univ_le
```

Comment:
It's a little bit weird to mix term mode and tactic mode like that, with two big `begin ... end` blocks. Would you go full tactic mode?

Reply:
Done, and it did indeed improve readability.